### ISSUE TITLE
Docker and install yarn does not work.

### ISSUE THREAD

**Issue by gabyx** [siouan/frontend-gradle-plugin#179]
**Description**
I am trying to installYarn but it does not work inside `docker.io/gabyxgabyx/technical-markdown:latest` docker container. Somehow `node` is not in the PATH environment??
This is from the project and branch:
https://github.com/gabyx/TechnicalMarkdown/blob/feature/docker-image/build.gradle.kts

Output:

```shell
╰─❯ ./gradlew -t build-html
Starting a Gradle Daemon, 1 busy and 1 stopped Daemons could not be reused, use --status for details

> Task :defineEnvironment
Checking executables ...
Pandoc version: 2.17.1.1
Pandoc exectuable found.

> Task :installNode

> Task :installYarnGlobally FAILED
env: can't execute 'node': No such file or directory

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':installYarnGlobally'.
> Process 'command '/bin/sh'' finished with non-zero exit value 127

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3m 42s
3 actionable tasks: 3 executed

Exiting continuous build as no executed tasks declared file system inputs.
```

**Environment**
 
- OS: Docker Image `docker.io/gabyxgabyx/technical-markdown:latest` Alpine.
- JDK: [e.g. Oracle JDK 11 64 bits]
- Gradle:
    ------------------------------------------------------------
    Gradle 7.0
    ------------------------------------------------------------
    
    Build time:   2021-04-09 22:27:31 UTC
    Revision:     d5661e3f0e07a8caff705f1badf79fb5df8022c4
    
    Kotlin:       1.4.31
    Groovy:       3.0.7
    Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
    JVM:          11.0.14 (Alpine 11.0.14+9-alpine-r0)
    OS:           Linux 5.10.76-linuxkit amd64

Settings in `build.gradle[.kts]` file:
```groovy
frontend {
    nodeVersion.set("17.7.1")
    nodeInstallDirectory.set(file("${project.buildDir}/node"))
    yarnEnabled.set(true)
    yarnVersion.set("1.22.18")
    packageJsonDirectory.set(file("${toolsDir}"))
    installScript.set("install '--modules-folder=${project.buildDir}/node_modules'")
}
```

**Comment by gabyx** [comment]
I guess I need yarn version "3.2.0". Lets try.

**Comment by gabyx** [comment]
Actually the same error with 3.2.0 which would be correct.

**Comment by gabyx** [comment]
I filed a bug report.
Maybe one could make a short regex/replacement hack for the wrong shebangs inside the node directory.

https://github.com/nodejs/node/issues/42367

**Comment by v1nc3n4** [comment]
Hi @gabyx,

Thank you for using the plugin.
After reading your comments, it appears to me the issue is more related to a change in Node.js distribution since release 15, where the shebang in `npm` executable has been updated. However, just to help, are you sure the Docker image you are using contains a `/usr/bin/env` executable, and that it supports `bash`?

Unless I miss something, I will close the issue. Feel free to open it again if you think there's something to be fixed in the plugin.
Have a good week-end!
BR

**Comment by gabyx** [comment]
Yes, the image supports env and bash. But the newest node.js cannot work as 
it trys to start npm over /usr/bin/env node which is weird…

how ever: can you add the node/bin directory to the environment when executing anythin inside this directory, that will nicely fix it! :)





Von meinem iPhone gesendet

**Comment by gabyx** [comment]
Yes, the image supports `env` and `bash`. But the newest node.js cannot work as 
it trys to start npm over `/usr/bin/env` node which is weird…

- Can you add the `.../node/bin` directory to the `PATH` environment when executing anything inside this
directory, that will nicely fix it for now, and is not a bad workaround?

**Comment by v1nc3n4** [comment]
Hello @gabyx,

The `bin` directory of the Node.js distribution should be already added to the `PATH` environment variable by the plugin just before the command is executed. You should see such information by running the plugin with the `verboseModeEnabled` flag and look for a line starting with `Execution settings: ...`.
If this is does not help, you may also use Gradle debug log level, which shall output the environment used for the command (line starting with `Execution environment: ...`).

Feel free to share these logs in case of misunderstanding.
Hope this helps!

**Comment by v1nc3n4** [comment]
Hi,
I'm closing the discussion because it doesn't seem to me there's a bug in the plugin. Feel free to reopen it with further details if needed.
Thank you!
BR

**Comment by v1nc3n4** [comment]
This [issue](https://github.com/siouan/frontend-gradle-plugin/issues/185) provides the solution to deal with Alpine O/S.
